src/NServiceBus.AcceptanceTests/Hosting/When_a_message_is_faulted.cs

```
@@ -0,0 +1,81 @@
+            Assert.IsNotNull(context.Machine, "Machine should be included in fault message headers.");
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool Done { get; set; }
+            public string HostId { get; set; }
+            public string HostName { get; set; }
+            public string Endpoint { get; set; }
+            public string Machine { get; set; }
+        }
+
+        public class EndpointWithAuditOn : EndpointConfigurationBuilder
+        {
+            public EndpointWithAuditOn()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.DisableFeature<Features.SecondLevelRetries>();
+                    c.SendFailedMessagesTo("errorQueueForAcceptanceTest");
+                });
+            }
+
+            public class MessageToBeAuditedHandler : IHandleMessages<MessageThatFails>
+            {
+                public Task Handle(MessageThatFails message, IMessageHandlerContext context)
+                {
+                    throw new SimulatedException();
+                }
+            }
+        }
+
+        class EndpointThatHandlesErrorMessages : EndpointConfigurationBuilder
+        {
+            public EndpointThatHandlesErrorMessages()
+            {
+                EndpointSetup<DefaultServer>().CustomEndpointName("errorQueueForAcceptanceTest");
+            }
+
+            public class MessageToBeAuditedHandler : IHandleMessages<MessageThatFails>
+            {
+                public Context TestContext { get; set; }
+
+                public Task Handle(MessageThatFails message, IMessageHandlerContext context)
+                {
+                    TestContext.HostId = context.MessageHeaders.ContainsKey(Headers.HostId) ? context.MessageHeaders[Headers.HostId] : null;
+                    TestContext.HostName = context.MessageHeaders.ContainsKey(Headers.HostDisplayName) ? context.MessageHeaders[Headers.HostDisplayName] : null;
```

@WojcikMike I agree with you. And since the headers can/could come from multiple locations it is cleaner to let the assertion fail instead of bubbling a `KeyNotFoundException`